### ISSUE TITLE
Documentation, version 2

### ISSUE THREAD

**Issue by nmansard** [stack-of-tasks/pinocchio#507]
I start this thread of discussion before implementing a new version of the documentation. I pushed in a new branch topic/doc-v2 (commit #29bcf0d) an outline that I suggest for the documentation. Please feedback on this proposition.

**Comment by nim65s** [comment]
I suggest that we add the section `Labs`, just after `Tutorials`, as we already have them, and keeping them up to date should be easy as you use those to teach classes.

**Comment by gabrielebndn** [comment]
I created a pull request with updates to the outline, as discussed with Nicolas.
I repeat the comment to the pull request here:

I think the doc should be modeled on Eigen's "long tutorial": a series of pages, each covering a topic from the user point of view, with worked examples in the form of short snippets intertwined with the code.
Then the proper tutorials will contain larger in-depth examples.
The technical part, containing implementation details, should be separate.
So, the main user doc in a part I called  "Features". As "List of features" should closely reflect the content of "Features", I removed all content from the former part and put in into the latter.
Also, I moved the math to later in the doc, as discussed
